segger: rtt: No multithreading support for rtt console

nordic-krch opened the pull request:
Updated rtt to not use mutexes when multithreading is disabled. Removed `k_sleep` call from rtt_console when multithreading is disabled. Those changes fixes the case when RTT console was failing to compile when multithreading was disabled.

nordic-krch committed:
drivers: console: rtt: Add no multithreading support

Avoid calling k_sleep when multithreading is disabled.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
manifest: update segger

Update segger with RTT version that support no multithreading.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nvlsianpu commented:
<!--StartFragment-->
/workdir/zephyr/subsys/logging/log_backend_rtt.c:158:6: error: unused variable 'ret' [-Werror=unused-variable]
--
&nbsp; | 158 \|  int ret = SEGGER_RTT_WriteSkipNoLock(CONFIG_LOG_BACKEND_RTT_BUFFER,
&nbsp; | \|      ^~~

<!--EndFragment-->

nvlsianpu reviewed (changes requested):
^^

nordic-krch committed:
logging: backend_rtt: Align to changes in RTT_LOCK/UNLOCK

RTT_LOCK/UNLOCK in certain configuration creates code block (curly
braces). In that case variables declared inside are local to that
block. Moved declaration of ret variable before the block. Updated
code to ensure that RTT_LOCK/UNLOCK are in the same code block.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>
